testbed/testbed/mock_backend.go

```diff
@@ -149,6 +159,59 @@ func (mb *MockBackend) ConsumeMetric(md pdata.Metrics) {
 	if mb.isRecording {
 		mb.ReceivedMetrics = append(mb.ReceivedMetrics, md)
 	}
+
+	// Record the timestamp of a metric from each scrape.
+	// Then the validator can check the difference between timestamps are the same as the scrape interval.
+	if mb.isRecordingMetricTimestamps {
+		currentTimestamp := getFirstMetricTimestamp(md)
+		receivedTimestampsCount := len(mb.ReceivedTimestamps)
+
+		// Record the timestamp only if it's the first recorded or if it's different from the last recorded timestamp
```

Comment:
Add a comment to explain why this is necessary. Why it matters that we only record the timestamps which are different from the last recorded. Why we record only the first timestamp if we receive several in this particular ConsumeMetric call.